motionplan/rrt.go

```diff
@@ -66,16 +64,55 @@ func (plan *rrtPlanReturn) err() error {
 type rrtMaps struct {
 	startMap rrtMap
 	goalMap  rrtMap
+	optNode  *costNode // The highest quality IK solution
 }
 
-func initRRTMaps() *rrtMaps {
-	return &rrtMaps{
-		startMap: map[node]node{},
-		goalMap:  map[node]node{},
+func initRRTMaps(ctx context.Context, mp motionPlanner, goal spatialmath.Pose, seed []referenceframe.Input) *rrtPlanReturn {
```

Comment:
probably should rename this something like checkInterpolateSolution (that was the first name I thought of I don't think its very good, but the new name should denote that this is doing more than initializing and is the first attempt we take to solve it)

Reply:
Indeed, since this is actually running the IK itself, which is the main function. The interpolate check is a secondary purpose.

How about "initRRTsolutions"?

Reply:
I think thats a little better. In a way the solution generation here is just a side effect of trying to interpolate which is the main purpose of the function.  Whatever you decide to go with you should add a comment on the function for clarity

Reply:
Done